### PR TITLE
PLANET-5078 Print duplicate metakey counts in CLI cmd before deleting

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
- Add new CLI cmd to print duplicate postmeta key counts (`wp p4-gblocks print_duplicate_postmeta`)
- On execution of `wp p4-gblocks remove_duplicate_postmeta` cmd print duplicate postmeta keys before delete. 
- Update Admin duplicate postmeta report with `meta_key` and counts

### CODE REVIEW

File: classes/command/class-duplicated-postmeta.php [hunk 1]
```diff
@@ -64,7 +64,29 @@ public static function detect() {
 				WHERE meta_key IN (" . self::generate_placeholders( self::META_KEY_LIST, 1 ) . ")
 				GROUP by `post_id`
 				HAVING all_count <> unique_count 
-				ORDER BY `all_count` DESC ";
+				ORDER BY `all_count` DESC";
+
+		$prepared_sql = $wpdb->prepare( $sql, self::META_KEY_LIST );
```
Comment (Inwerpsel) [inline classes/command/class-duplicated-postmeta.php:69]: This prepare is probably a leftover as there is no param anymore.

File: classes/command/class-duplicated-postmeta.php [hunk 2]
```diff
@@ -64,7 +64,29 @@ public static function detect() {
...
+		global $wpdb;
+
+		// phpcs:disable
+		$sql = "SELECT `meta_key`, COUNT(post_id) AS all_count , COUNT(DISTINCT post_id) AS unique_count
+				FROM `wp_postmeta`
+				GROUP by `meta_key`
+				HAVING all_count <> unique_count
+				ORDER BY `all_count` DESC";
 
 		$prepared_sql = $wpdb->prepare( $sql, self::META_KEY_LIST );
```
Comment (Inwerpsel) [inline classes/command/class-duplicated-postmeta.php:91]: This doesn't have the parameter `self::META_KEY_LIST`

File: classes/command/class-duplicated-postmeta.php [hunk 3]
```diff
@@ -64,7 +64,29 @@ public static function detect() {
...
+		return $wpdb->get_results( $prepared_sql );
+		// phpcs:enable
+	}
+
+	/**
+	 * Detect duplicate postmeta meta_key's with count
+	 *
+	 * @return object
+	 */
+	public static function detect_metakey_counts() {
```
Comment (Inwerpsel) [inline classes/command/class-duplicated-postmeta.php:80]: Can't this use the same function as for the admin page? Probably there it makes sense too to detect all meta keys.

File: classes/command/class-duplicated-postmeta.php (hunk 2 above)
Comment (sagarsdeshmukh) [reply classes/command/class-duplicated-postmeta.php:91]: ops, my bad. good catch 👍

File: classes/command/class-duplicated-postmeta.php (hunk 3 above)
Comment (sagarsdeshmukh) [reply classes/command/class-duplicated-postmeta.php]: I updated the same logic for admin report and keep single detect method. Please have a look.

File: classes/command/class-controller.php [hunk 4]
```diff
@@ -66,12 +66,14 @@ public function remove_duplicate_postmeta() {
...
+
+			$this->print_duplicate_postmeta();
 
 			$deleted_rows = Duplicated_Postmeta::remove();
 
 			if ( $deleted_rows ) {
-				WP_CLI::success( "Removed $deleted_rows duplicate postmeta record/s" );
+				WP_CLI::success( "Removed $deleted_rows duplicate postmeta records" );
 			} else {
 				WP_CLI::log( 'No duplicate postmeta record found.' );
```
Comment (Inwerpsel) [inline classes/command/class-controller.php:78]: Maybe you can make this "Now whitelisted duplicate postmeta records found"?

**Pull request merged**